cpp/util/read_private_key.cc

```diff
@@ -24,5 +27,68 @@ KeyError ReadPrivateKey(EVP_PKEY** pkey, const std::string& file) {
 }
 
 
+static ENGINE* engine = NULL;
```

Comment:
Pretty much everything that touches this is crying out for a wrapper class... All those ```atexit```, eek!